app/models/piece.rb

```diff
@@ -19,6 +19,20 @@ def move_to!(new_row, new_col)
     update(row_position: new_row, col_position: new_col, moved: true)
   end
 
+  def moving_into_check?(row_dest, col_dest)
+    row_pos = row_position
```

Comment:
Yes, this is why I used those variables :)
I'll try assign_attributes and see what happens. 
Regarding Metrics/LineLength - I tried disabling it below the class, and also above specific lines, but somehow it didn't work. I'll try again.